datastream-common/src/main/java/com/linkedin/datastream/common/BrooklinEnvelope.java

```diff
@@ -48,11 +51,25 @@ public BrooklinEnvelope(Object key, Object value, Map<String, String> metadata)
    */
   public BrooklinEnvelope(@Nullable Object key, @Nullable Object value, @Nullable Object previousValue,
       Map<String, String> metadata) {
+    this(key, value, previousValue, null, metadata);
+  }
+
+  /**
+   * Construct a {@link BrooklinEnvelope} using record key, value, headers and metadata
+   * @param key The record key (e.g. primary key)
+   * @param value The new record value
+   * @param previousValue The old record value
+   * @param headers Kafka headers to associate with the change event
+   * @param metadata Additional metadata to associate with the change event
+   */
+  public BrooklinEnvelope(@Nullable Object key, @Nullable Object value, @Nullable Object previousValue,
+      @Nullable Iterable<Header> headers, Map<String, String> metadata) {
     Validate.notNull(metadata, "metadata cannot be null");
     setKey(key);
     setValue(value);
     setPreviousValue(previousValue);
     setMetadata(metadata);
+    setHeaders(headers);
```

Comment:
nit: can we keep order of function parameters the same as order of setting them? so here move this before setMetadata(metadata)?